core/iwasm/common/wasm_runtime_common.c

```diff
@@ -863,6 +863,183 @@ wasm_runtime_call_wasm(WASMExecEnv *exec_env,
+                out_argv += 2;
+                p += 2;
+                break;
+        }
+    }
+    return p;
+}
+
+uint32 parse_uint32_array_to_results(WASMType *type,
+                                uint32 argc, uint32 *argv,
+                                wasm_val_t *out_results)
+{
+    int i, p;
+
+    for (i = 0, p = 0; i < type->result_count; i++) {
+        switch (type->types[type->param_count + i]) {
+            case VALUE_TYPE_I32:
+                out_results[i].kind = WASM_I32;
+                out_results[i].of.i32 = *(int32 *)argv;
+                argv++;
+                p++;
+                break;
+            case VALUE_TYPE_I64:
+                out_results[i].kind = WASM_I64;
+                out_results[i].of.i64 = *(int64 *)argv;
+                argv += 2;
+                p += 2;
+                break;
+            case VALUE_TYPE_F32:
+                out_results[i].kind = WASM_F32;
+                out_results[i].of.f32 = *(float32 *)argv;
+                argv++;
+                p++;
+                break;
+            case VALUE_TYPE_F64:
+                out_results[i].kind = WASM_F64;
+                out_results[i].of.f64 = *(float64 *)argv;
+                argv += 2;
+                p += 2;
+                break;
+        }
+    }
+    bh_assert(argc == p);
+    return type->result_count;
+}
+
+bool
+wasm_runtime_call_wasm_a(WASMExecEnv *exec_env,
+                       WASMFunctionInstanceCommon *function,
+                       uint32 num_results, wasm_val_t results[],
+                       uint32 num_args, wasm_val_t args[])
+{
+    uint32 argc, *argv, ret_num, cell_num, total_size;
+    bool ret = false;
+    WASMType *type = NULL;
+
+#if WASM_ENABLE_INTERP != 0
+    if (exec_env->module_inst->module_type == Wasm_Module_Bytecode) {
+        WASMFunctionInstance *wasm_func = (WASMFunctionInstance*)function;
+        type = wasm_func->u.func->func_type;
+        argc = wasm_func->param_cell_num;
+        cell_num = argc > wasm_func->ret_cell_num ?
+                   argc : wasm_func->ret_cell_num;
+    }
+#endif
+#if WASM_ENABLE_AOT != 0
+    if (exec_env->module_inst->module_type == Wasm_Module_AoT) {
+        type = ((AOTFunctionInstance*)function)->u.func.func_type;
+        argc = type->param_cell_num;
+        cell_num = argc > type->ret_cell_num ?
+                   argc : type->ret_cell_num;
+    }
+#endif
+
+    if (num_results != type->result_count || num_args != type->param_count)
+        goto fail2;
+
+    total_size = sizeof(uint32) * (uint64)(cell_num > 2 ? cell_num : 2);
+    if (!(argv = runtime_malloc((uint32)total_size, exec_env->module_inst, NULL, 0)))
+        goto fail2;
+
+    argc == parse_args_to_uint32_array(type, num_args, args, argv);
+    if (!(ret = wasm_runtime_call_wasm(exec_env, function, argc, argv)))
+        goto fail1;
+    ret_num = parse_uint32_array_to_results(type, type->ret_cell_num, argv, results);
+    if (ret_num != num_results) {
```

Comment:
Use bh_assert(ret_num == num_results), they should be same, if not ,it should be runtime's bug